src/include/storage/projected_columns.h

```diff
@@ -204,12 +201,19 @@ class PACKED ProjectedColumns {
                                                          common::RawBitmap::SizeInBytes(max_tuples_));
   }
 
+  /**
+   * @param col_id the column ID within the projection we want the size for
+   * @return the size (in bytes) of the attributes in this column
+   */
+  const uint32_t AttrSizeForColumn(uint16_t col_id);
+
  private:
   friend class ProjectedColumnsInitializer;
   uint32_t size_;
   uint32_t max_tuples_;
   uint32_t num_tuples_;
   uint16_t num_cols_;
+  uint16_t attr_ends_[4];
```

Comment:
Why is this 4? I think we need a comment explaining what each of these mean and why it has size 4. It's not immediately obvious.

Reply:
Replaced all uses of the magic 4 with a defined constant, and added comments there and in `type_util.h` about the power of two assumption on attribute sizes (@mbutrovich recommended this).

Reply:
Replaced the magic number with a constant definition and added comments to explain how we arrived at that comment.  I also put a warning in the type utils where attribute sizes are defined that this constant exists in the storage layer.